src/services/ExpressionBuilder.ts

```diff
@@ -603,22 +623,31 @@ export class ExpressionBuilder {
    * Groups all filters by operator and key
    */
   private groupFiltersByKey(filters: AdHocVariableFilter[]): Record<FilterOpType, Dictionary<AdHocFilterWithLabels[]>> {
-    const positiveMatch = filters.filter(
+    let filteredFilters: AdHocVariableFilter[] = filters.filter((f) => !this.options.ignoreKeys?.includes(f.key));
+
+    // We need at least one inclusive filter
+    if (this.options.type === 'indexed') {
+      if (filteredFilters.length < 1) {
+        filteredFilters = filters;
```

Comment:
So if we have multiple include filters containing the exclude key, we would run the query with all of them. Maybe this is a reason to exclude the "primary" filter from breakdown views so we know that we're never excluding the only include filter.

But folks might want write a regex include e.g. `cluster=~"us-.+"`, and then want to exclude stuff in the UI... but right now I believe this will behave as the current version, including will hide all other options, and excluding will remove that panel from the view (if they have no other inclusion label filters).